quartz/plugins/emitters/contentIndex.ts

```diff
@@ -116,50 +121,67 @@ export const ContentIndex: QuartzEmitterPlugin<Partial<Options>> = (opts) => {
-        if (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) {
-          linkIndex.set(slug, {
-            title: file.data.frontmatter?.title!,
-            links: file.data.links ?? [],
-            tags: file.data.frontmatter?.tags ?? [],
-            content: file.data.text ?? "",
-            richContent: opts?.rssFullHtml
-              ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
-              : undefined,
-            date: date,
-            description: file.data.description ?? "",
-          })
+      const feedIndices: Map<String, ContentIndex> = new Map()
+
+      // bfahrenfort: ts can't see the expansion of opts above that guarantees a non-null feedDirectories
+      const directories =
+        opts?.feedDirectories == null ? defaultOptions.feedDirectories : opts.feedDirectories
+
+      for (const feed of directories) {
+        const linkIndex: ContentIndex = new Map()
+        for (const [tree, file] of content) {
+          const slug = file.data.slug!
+
+          const date = getDate(ctx.cfg.configuration, file.data) ?? new Date()
+          if (
+            (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) &&
+            (slug.startsWith(feed) || feed == "index")
+          ) {
+            linkIndex.set(slug, {
+              title: file.data.frontmatter?.title!,
+              links: file.data.links ?? [],
+              tags: file.data.frontmatter?.tags ?? [],
+              content: file.data.text ?? "",
+              richContent: opts?.rssFullHtml
+                ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
+                : undefined,
+              date: date,
+              description: file.data.description ?? "",
+            })
+          }
         }
+        feedIndices.set(feed, linkIndex)
       }
 
       if (opts?.enableSiteMap) {
         emitted.push(
           await write({
             ctx,
-            content: generateSiteMap(cfg, linkIndex),
+            // bfahrenfort: "index" is guaranteed non-null
+            // see directories instantiation and feedIndices.set iterating over directories
+            content: generateSiteMap(cfg, feedIndices.get("index")!),
             slug: "sitemap" as FullSlug,
             ext: ".xml",
           }),
         )
       }
 
       if (opts?.enableRSS) {
-        emitted.push(
-          await write({
+        directories.map(async (feed) => {
+          const emittedFeed = await write({
             ctx,
-            content: generateRSSFeed(cfg, linkIndex, opts.rssLimit),
-            slug: "index" as FullSlug,
+            // bfahrenfort: we just generated a feedIndices entry for every directories entry, guaranteed non-null
+            content: generateRSSFeed(cfg, feedIndices.get(feed)!, opts?.rssLimit),
+            slug: feed as FullSlug,
             ext: ".xml",
-          }),
-        )
+          })
+          emitted.push(emittedFeed)
+        })
       }
```

Comment:
I think is is anti-pattern. Best to create an array of promises and then `Promise.all` it

Reply:
This is me not understanding async-await parallelism haha. Thanks for the tip

Reply:
You can check out https://github.com/jackyzha0/quartz/blob/21c6bbf3026cf9c06f6458bdb6614099d9a91adc/quartz/plugins/emitters/componentResources.ts#L195 downwards for an example how to do this.

Reply:
I'm surprised that the compiler doesn't throw an error here. I thought `map` was a sync operation.